Feature/14 service to get performance data

TheTanic opened the pull request:
Handles Issue #14 
Adds support of CPU-Utilization tracking and RAM-Usage. 
The other performance indicator are postponed to a later time. 
Inside of the DashboardPageViewModel is an example how to register for the events, which hold the informations about the performance indicators. 
The solution is already aware of different plattforms. Currently only Windows is supported.

TheTanic committed:
Add first verison of PerformanceDataService

TheTanic committed:
Cleanup the creation of the PerformanceCounter for CPU utilization

TheTanic committed:
Test commit with warnings, to see if the build fails

TheTanic committed:
Made PerformanceData collection platform specific.

TheTanic committed:
Reogranize Extension

TheTanic committed:
Merge remote-tracking branch 'origin/main' into feature/14-service-to-get-performance-data

TheTanic committed:
Remove CPU-Temperature for now

TheTanic committed:
Merge branch 'main' into feature/14-service-to-get-performance-data

TheTanic commented on src/Semoda/Semoda/Services/PerformanceDataService.cs:
```diff
@@ -0,0 +1,83 @@
+        /// <inheritdoc/>
+        public Task<bool> DeregisterAsync(Action<object, PerformanceDataEventArgs> eventHandler)
+        {
+            throw new NotImplementedException();
```
This method is not implemented for now, because there is not a use case yet

TheTanic commented on src/Semoda/Semoda/ViewModels/DashboardPageViewModel.cs:
```diff
@@ -13,6 +18,25 @@ public class DashboardPageViewModel : ViewModelBase
         /// </summary>
         public DashboardPageViewModel() : base(true)
         {
+            IPerformanceDataService dataService = ServiceProvider.GetRequiredService<IPerformanceDataService>();
```
This is just a show off of the functionality. Will be removed, when there is a view to represent the data

Ploinky commented on src/Semoda/Semoda/ViewModels/DashboardPageViewModel.cs:
```diff
@@ -13,6 +18,25 @@ public class DashboardPageViewModel : ViewModelBase
+
+        private void HandleCPUUtilization(object? sender, PerformanceDataEventArgs args)
+        {
+            if (args.PerformanceDataType == Models.PerformanceDataType.TotalCPUUtilization)
```
seems inelegant to have to do this if we're already registering 2 different methods to catch the 2 different events, but i don't know enough about the language and framework to tell if there's a better way.

TheTanic replied:
We need to register twice, because every performance counter handles a specific parameter. But indeed it is possible to merge both HandleMethods to one method. 
The check which kind of performance data is collected needs to be done, because every performance data runs through the same event bus. Otherwise we would need to have a single EventHandler for every possible future data type.

TheTanic committed:
Handle PR:
- Corrected typos
- Merged HandleData Methods
- Prepare polling delay for future configuration

TheTanic commented on src/Semoda/Semoda/Services/PerformanceDataService.cs:
```diff
@@ -0,0 +1,83 @@
+            {
+                while (!_cts.IsCancellationRequested)
+                {
+                    await Task.Delay(1000);
```
The delay moved to a variable, to be configurable in the future. The defaul is still 1000 ms

Ploinky commented on src/Semoda/Semoda/Services/Interfaces/IPerformanceDataService.cs:
```diff
@@ -0,0 +1,40 @@
+    public interface IPerformanceDataService
+    {
+        /// <summary>
+        /// Deregister at the service, to no longet get notified about new events.
```
`longet` 😆
